src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,170 @@
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkBounds(index);
+        Node<T> node = getNodeByIndex(index);
+        return node.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkBounds(index);
+        Node<T> node = getNodeByIndex(index);
+        T lastValue = node.value;
+        node.value = value;
+        return lastValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkBounds(index);
+        T value = getNodeByIndex(index).value;
+        unlink(getNodeByIndex(index));
+        return value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if (object == currentNode.value
+                    || (object != null && object.equals(currentNode.value))) {
+                unlink(currentNode);
+                return true;
+            }
+            currentNode = currentNode.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(Node<T> node) {
+        if (node == head || (node != null && node.equals(head))) {
+            removeFirst();
+            return;
+        }
+        if (node == tail || (node != null && node.equals(tail))) {
+            removeLast();
+            return;
+        }
+        Node<T> prevNode = node.prev;
+        Node<T> nextNode = node.next;
+        nextNode.prev = prevNode;
+        prevNode.next = nextNode;
+        size--;
+    }
+
+    private void addFirst(Node<T> node) {
+        head.prev = node;
+        node.next = head;
+        head = node;
+        size++;
+    }
+
+    private void removeFirst() {
+        Node<T> currentNode = head.next;
+        currentNode.prev = null;
+        head = currentNode;
+        size--;
     }
+
+    private void removeLast() {
+        Node<T> currentNode = tail.prev;
+        currentNode.next = null;
+        tail = currentNode;
+        size--;
+    }
+
+    private void checkBounds(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("The index: " + index + " is out of bounds");
+        }
+    }
+
+    private Node<T> getNodeByIndex(int index) {
+        Node<T> node;
+        if (index >= (double) size / 2) {
+            node = tail;
+            for (int i = size - 1; i > index; i--) {
+                node = node.prev;
+            }
+            return node;
+        }
+        if (index < (double) size / 2) {
+            node = head;
+            for (int i = 0; i < index; i++) {
+                node = node.next;
+            }
+            return node;
+        }
+        return null;
```

Comment:
Why do you return null? This method shouldn't return null, cause you should check bounds before calling this method.